### ISSUE TITLE
[route53] Support for health checks

### ISSUE THREAD

**Issue by justin8** [aws/aws-cdk#9481]
Currently the only way to define a health check or even to enable health checks for alias targets appears to be using `Cfn` resources instead of the higher level constructs.

### Use Case
Route53 health checks are commonly used for multi-region architectures, and not being able to create them is a little limiting

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by shivlaks** [comment]
@justin8 I agree this is a pretty important missing feature. marking it as a `p1` to help us prioritize getting it implemented and added in.

**Comment by elhedran** [comment]
I see this has been opened for a while now.  Can someone at put the work around using `Cfn` resources while we wait for a fix?

**Comment by elhedran** [comment]
Trying to work around this open ticket... not sure this is right though.  Deployment seems to get stuck.

```Typescript
const alias = new route53_targets.ApiGatewayv2DomainProperties(
    domainName.regionalDomainName, domainName.regionalHostedZoneId
)
const target = route53.RecordTarget.fromAlias(alias)

// latency based routing with health check
const gwStack = core.Stack.of(gw)
const healthCheck = new route53.CfnHealthCheck(this, 'healthCheck', {
    healthCheckConfig: {
        fullyQualifiedDomainName:
            `${gw.apiId}.execute-api.${gwStack.region}.${gwStack.urlSuffix}`,
        type: 'HTTPS',
        measureLatency: true
    }
})

new route53.CfnRecordSet(this, 'aRecordSet', {
    aliasTarget: {
        dnsName: domainName.regionalDomainName,
        hostedZoneId: domainName.regionalHostedZoneId,
        evaluateTargetHealth: true
    },
    name: siteName,
    hostedZoneId: hostedZone.hostedZoneId,
    resourceRecords: target.values,
    type: route53.RecordType.A,
    region: gwStack.region,
    setIdentifier: id,
    healthCheckId: healthCheck.attrHealthCheckId
})
```

I'll update if I work out something better, but wanted to put what I had so far down given this has been open a while now

**Comment by Fitmavincent** [comment]
Before this feature is available in the new release, do we have any way to associate a Cloudwatch Alarm with the Route53 Health Check in CDK?

**Comment by OGoodness** [comment]
Any update on this?

**Comment by pflorek** [comment]
Here is a small custom construct that should work with the higher level `RecordSet`. Supports Endpoint, Alarm and Calculated health checks. Has method to retrieve metrics for CloudWatch Alarms and methods to configure failover.

[cdk-route53-health-check](https://github.com/pepperize/cdk-route53-health-check)
![diagram](https://user-images.githubusercontent.com/13916107/197164478-92c4209d-139d-46da-8661-43236bef8b0b.png)

**Comment by wladyslawczyzewski** [comment]
for those who are waiting for this: I have already opened the PR for alias records health checks and now I'm also working on regular health checks expecting the PR to be ready in a few days 🤞

**Comment by wladyslawczyzewski** [comment]
two PRs are waiting for review & merge – first one for alias target health evaluation & second one for `AWS::Route53::HealthCheck` resources

**Comment by wladyslawczyzewski** [comment]
both PRs merged - health checks are now available in AWS CDK
